index_search_test.go

```diff
@@ -916,3 +916,172 @@ func TestIndex_SearchWithSort(t *testing.T) {
 		})
 	}
 }
+
+func TestIndex_SearchOnNestedFileds(t *testing.T) {
+	type args struct {
+		UID                 string
+		PrimaryKey          string
+		client              *Client
+		query               string
+		request             SearchRequest
+		searchableAttribute []string
+		sortableAttribute   []string
+	}
+	tests := []struct {
+		name string
+		args args
+		want *SearchResponse
+	}{
+		{
+			name: "TestIndexBasicSearchOnNestedFields",
+			args: args{
+				UID:     "TestIndexBasicSearchOnNestedFields",
+				client:  defaultClient,
+				query:   "An awesome",
+				request: SearchRequest{},
+			},
+			want: &SearchResponse{
+				Hits: []interface{}{
+					map[string]interface{}{
+						"id": float64(5), "title": "The Hobbit",
+					},
+				},
+				NbHits:           1,
+				Offset:           0,
+				Limit:            20,
+				ExhaustiveNbHits: false,
+			},
+		},
+		{
+			name: "TestIndexBasicSearchOnNestedFieldsWithCustomClient",
+			args: args{
+				UID:     "TestIndexBasicSearchOnNestedFieldsWithCustomClient",
+				client:  customClient,
+				query:   "An awesome",
+				request: SearchRequest{},
+			},
+			want: &SearchResponse{
+				Hits: []interface{}{
+					map[string]interface{}{
+						"id": float64(5), "title": "The Hobbit",
+					},
+				},
+				NbHits:           1,
+				Offset:           0,
+				Limit:            20,
+				ExhaustiveNbHits: false,
+			},
+		},
+		{
+			name: "TestIndexSearchOnMultipleNestedFields",
+			args: args{
+				UID:     "TestIndexSearchOnMultipleNestedFields",
+				client:  defaultClient,
+				query:   "french",
+				request: SearchRequest{},
+			},
+			want: &SearchResponse{
+				Hits: []interface{}{
+					map[string]interface{}{
+						"id": float64(2), "title": "Le Petit Prince",
+					},
+					map[string]interface{}{
+						"id": float64(3), "title": "Le Rouge et le Noir",
+					},
+				},
+				NbHits:           2,
+				Offset:           0,
+				Limit:            20,
+				ExhaustiveNbHits: false,
+			},
+		},
+		{
+			name: "TestIndexSearchOnNestedFieldsWithSearchableAttribute",
+			args: args{
+				UID:     "TestIndexSearchOnNestedFieldsWithSearchableAttribute",
+				client:  defaultClient,
+				query:   "An awesome",
+				request: SearchRequest{},
+				searchableAttribute: []string{
+					"title", "info.comment",
+				},
+			},
+			want: &SearchResponse{
+				Hits: []interface{}{
+					map[string]interface{}{
+						"id": float64(5), "title": "The Hobbit",
+					},
+				},
```

Comment:
should you test the nested object as well?

Reply:
Done !